app/js/script.js

```diff
@@ -0,0 +1,160 @@
+  const timeZoneDate = new Date(utcDate.toLocaleString('en-US', { timeZone: selectedTimeZone }));
+
+  lastTime.seconds = timeZoneDate.getSeconds();
+  lastTime.minutes = timeZoneDate.getMinutes();
+  lastTime.hours = timeZoneDate.getHours() % 12;
+
+  updateClock();
+}
+
+function startDragging(event) {
+  dragging = true;
+  this.classList.add('dragging');
+  clearInterval(clockInterval);
+}
+
+function handleDragging(event) {
+  if (dragging) {
+    const draggingHand = document.querySelector('.dragging');
+    if (draggingHand) {
+      setRotationAngle(draggingHand, event);
+    }
+  }
+}
+
+function stopDragging() {
+  dragging = false;
+  document.querySelectorAll('.dragging').forEach(hand => hand.classList.remove('dragging'));
+  setTimeBasedOnHandPositions();
+  clockInterval = setInterval(updateClock, 1000);
+}
+
+function getAngle(el) {
+    const svg = document.querySelector('svg');
+    const centerX = svg.getBoundingClientRect().left + svg.clientWidth / 2;
+    const centerY = svg.getBoundingClientRect().top + svg.clientHeight / 2;
+  
+    const rect = el.getBoundingClientRect();
+    const x = rect.left + rect.width / 2;
+    const y = rect.top + rect.height / 2;
+  
+    let angle = 90 + (Math.atan2(y - centerY, x - centerX) * 180) / Math.PI;
+    if (angle < 0){
+        angle += 360;
+    }
+    return angle;
+  }
+
+function setRotationAngle(hand, event) {
+  const svg = document.querySelector('svg');
+  const centerX = svg.getBoundingClientRect().left + svg.clientWidth / 2;
+  const centerY = svg.getBoundingClientRect().top + svg.clientHeight / 2;
+
+  let angle = 90 + Math.atan2(event.clientY - centerY, event.clientX - centerX) * (180 / Math.PI);
+  if (angle < 0) {
+    angle += 360;
+  }
+
+  hand.style.transform = `rotate(${angle}deg)`;
+}
+
+function setTimeBasedOnHandPositions() {
+    const hourAngle = getAngle(hourHand);
+    const minuteAngle = getAngle(minuteHand);
+    const secondAngle = getAngle(secondHand);
+  
+    const hours = Math.floor(hourAngle / 30) % 12;
+    const minutes = Math.floor(minuteAngle / 6);
+    const seconds = Math.floor(secondAngle / 6);
+  
+    lastTime.hours = hours;
+    lastTime.minutes = minutes;
+    lastTime.seconds = seconds;;
+  }
+
+function updateClock() {
+  if (!dragging) {
+    const currDate = new Date();
+    const utcDate = new Date(currDate.toLocaleString('en-US', { timeZone: 'UTC' }));
+    const timeZoneDate = new Date(utcDate.toLocaleString('en-US', { timeZone: selectedTimeZone }));
+
+    lastTime.seconds = timeZoneDate.getSeconds();
+    lastTime.minutes = timeZoneDate.getMinutes();
+    lastTime.hours = timeZoneDate.getHours() % 12;
+
+    const seconds = lastTime.seconds / 60;
+    const minutes = (lastTime.minutes + seconds) / 60;
+    const hours = (lastTime.hours + minutes) / 12;
+
+    setRotation(secondHand, seconds);
+    setRotation(minuteHand, minutes);
+    setRotation(hourHand, hours);
```

Comment:
The function setRotation() is called before it is defined. Move its definition before the call.